js/ace.js

```diff
@@ -0,0 +1,1021 @@
+            },
+            'requiredCredentials': {
+                'apiKey': true,
+                'secret': true,
+                'phone': true,
+            },
+            'api': {
+                'public': {
+                    'get': [
+                        'oapi/v2/list/tradePrice',
+                        'oapi/v2/list/marketPair',
+                        'open/v2/public/getOrderBook',
+                    ],
+                },
+                'private': {
+                    'post': [
+                        'v2/coin/customerAccount',
+                        'v2/kline/getKline',
+                        'v2/order/order',
+                        'v2/order/cancel',
+                        'v2/order/getOrderList',
+                        'v2/order/showOrderStatus',
+                        'v2/order/showOrderHistory',
+                        'v2/order/getTradeList',
+                    ],
+                },
+            },
+            'fees': {
+                'trading': {
+                    'percentage': true,
+                    'maker': this.parseNumber ('0.0005'),
+                    'taker': this.parseNumber ('0.001'),
+                },
+            },
+            'options': {
+                'networks': {
+                    'ERC20': 'ERC20',
+                    'ETH': 'ERC20',
+                    'TRX': 'TRX',
+                    'TRC20': 'TRX',
+                },
+            },
+            'precisionMode': TICK_SIZE,
+            'exceptions': {
+                'exact': {
+                    '2021': InsufficientFunds,
+                    '2061': BadRequest,
+                    '9996': BadRequest,
+                    '20182': AuthenticationError,
+                },
+                'broad': {
+                },
+            },
+            'commonCurrencies': {
+            },
+        });
+    }
+
+    async fetchMarkets (params = {}) {
+        /**
+         * @method
+         * @name ace#fetchMarkets
+         * @description retrieves data on all markets for ace
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[object]} an array of objects representing market data
+         */
+        const response = await this.publicGetOapiV2ListMarketPair ();
+        //
+        //     [
+        //         {
+        //             "symbol":"BTC/USDT",
+        //             "base":"btc",
+        //             "baseCurrencyId": "122"
+        //             "quote":"usdt",
+        //             "basePrecision":"8",
+        //             "quotePrecision":"5",
+        //             "minLimitBaseAmount":"0.1",
+        //             "maxLimitBaseAmount":"480286"
+        //         }
+        //     ]
+        //
+        const result = [];
+        for (let i = 0; i < response.length; i++) {
+            const market = response[i];
+            const base = this.safeString (market, 'base');
```

Comment:
we should unify the base and quote with `safeCurrencyCode` so the user can resolve conflicts with `commonCurrencies` if needed